### PR TITLE
add shift-parser

### PR DESCRIPTION
**Pull request by hzoo**
I got the same error as when add babel-eslint so doing something wrong here?

### CODE REVIEW

**Comment by fkling** [comment]
I just added a fix that should help to figure out what is wrong. shift doesn't have a `parse` method. It's either `parseScript` or `parseModule`, `parser.parse` throws. There might be a similar problem with babel-eslint.

**Comment by fkling** [comment]
(also thank you for doing this :) )

**Comment by hzoo** [comment]
Well babel-eslint definetely does https://github.com/babel/babel-eslint/blob/master/index.js#L367 :)

**Comment by fkling** [comment]
Might be some other runtime error inside the module then.

**Comment by hzoo** [comment]
Ok! That fixed it for me - called either module/script based on the option and it works.

File: src/parsers/shift.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import React from 'react'; // eslint-disable-line no-unused-vars
+import pkg from 'shift-parser/package.json';
+import loadAndExecute from './utils/loadAndExecute';
```
Comment (hzoo) [inline src/parsers/shift.js]: Also I just noticed that the other parsers have `loadAndExectue`

Comment (fkling) [inline src/parsers/shift.js]: Not sure what you mean?

Comment (hzoo) [inline src/parsers/shift.js]: Just spelling not a big deal https://github.com/fkling/esprima_ast_explorer/blob/cbf1b2da0339c6e4af317a8ae462731e840346d9/src/parsers/acorn.js#L3

Comment (fkling) [inline src/parsers/shift.js]: Oh, I didn't notice the typo :P

**Comment by hzoo** [comment]
@fkling this good to merge now?

**Comment by fkling** [comment]
Probably, just busy atm ;)

**Comment by hzoo** [comment]
Oh gotta update these PRs then

**Comment by fkling** [comment]
Merged with 3549372bf996e3b3c8914e952e6b45d33ebb3390 !

**Comment by RReverser** [comment]
@hzoo Did Shift support locations directly on Nodes back when you added it? Looks like it's not working, and documentation states completely separate interface for that.

**Pull request closed without merging**